src/main/java/reactor/ipc/netty/http/HttpOperations.java

```diff
@@ -187,4 +244,24 @@ protected final boolean markHeadersAsSent() {
 			AtomicIntegerFieldUpdater.newUpdater(HttpOperations.class,
 					"statusAndHeadersSent");
 
+	final static ChannelInboundHandler HTTP_EXTRACTOR = NettyPipeline.inboundHandler(
+			(ctx, msg) -> {
+				if (msg instanceof ByteBufHolder) {
+					ByteBuf bb = ((ByteBufHolder) msg).content();
+					if(msg instanceof FullHttpMessage){
+						// TODO convert into 2 messages if FullHttpMessage
+						ctx.fireChannelRead(msg);
+					}
+					else {
+						ctx.fireChannelRead(bb);
+						if (msg instanceof LastHttpContent) {
```

Comment:
... `HttpExtractorHandler` would only do that if for the lambda, the input was `LastHttpContent` and the result wasn't.